tests/test_authentication.py

```diff
@@ -140,13 +142,19 @@ def test_unknown_creds(self):
                               '^access denied: CredentialsLookupError: ')
 
     def test_alternative_credential_lookup(self):
-        sender = self._sender(credentials=ALTERNATIVE_CREDS)
-        req = self._request(sender)
         lookup_path = "%s.alternative_lookup" % __name__
         with self.settings(HAWK_CREDENTIALS_LOOKUP=lookup_path):
+            # Reload with settings change, re-instantiate
+            reload_module(hawkrest)
```

Comment:
I'd rather not see `reload_module()` used in the tests because it can introduce subtle bugs (e.g. references in Python are hard to keep track of). If you introduce a utility function like `get_credentials_lookup()` that returns a callable then you could probably use the same `with self.settings(...)` block without needing to reload the module.